runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java

```diff
@@ -116,6 +132,62 @@ public void onConfigChanged(ConfigChanged event)
 		}
 	}
 
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		int groupId = WidgetInfo.TO_GROUP(event.getActionParam1());
+		String option = event.getOption();
+
+		if (groupId == WidgetInfo.FRIENDS_LIST.getGroupId() || groupId == WidgetInfo.CLAN_CHAT.getGroupId() ||
+				groupId == WidgetInfo.CHATBOX.getGroupId() && !KICK_OPTION.equals(option) || //prevent from adding for Kick option (interferes with the raiding party one)
+				groupId == WidgetInfo.RAIDING_PARTY.getGroupId() || groupId == WidgetInfo.PRIVATE_CHAT_MESSAGE.getGroupId())
+		{
+			final MenuEntry lookup = new MenuEntry();
+			lookup.setOption(LOOKUP);
+			lookup.setTarget(event.getTarget());
+			lookup.setType(MenuAction.RUNELITE.getId());
+			lookup.setParam0(event.getActionParam0());
+			lookup.setParam1(event.getActionParam1());
+
+			if (BEFORE_OPTIONS.contains(option))
+			{
+				addMenuEntry(option, lookup, client.getMenuEntries(), false);
+			}
+			else if (AFTER_OPTIONS.contains(option))
+			{
+				addMenuEntry(option, lookup, client.getMenuEntries(), true);
+			}
+		}
+	}
+
+	private void addMenuEntry(String option, MenuEntry newEntry, MenuEntry[] entries, boolean after)
```

Comment:
This is always going to be in one of the last two slots... since the hook is called as the menu is built.